src/main/java/com/microsoft/aad/adal4j/AuthenticationContext.java

```diff
@@ -255,6 +253,54 @@ public AuthenticationResult call() throws Exception {
     }
 
     /**
+     * Acquires a security token from authority using an authenticated client with password credential
+     * grant.
+     *
+     * @param resource
+     *            Identifier of the target resource that is the recipient of the
+     *            requested token. If null, token is requested for the same
+     *            resource refresh token was originally issued for. If passed,
+     *            resource should match the original resource used to acquire
+     *            refresh token unless token service supports refresh token for
+     *            multiple resources.
+      @param credential
+     *            object representing Private Key to use for token acquisition.
+     * @param username
+     *            Username of the managed or federated user.
+     * @param password
+     *            Password of the managed or federated user.
+     * @param callback
+     *            optional callback object for non-blocking execution.
+     * @return A {@link Future} object representing the
+     *         {@link AuthenticationResult} of the call. It contains Access
+     *         Token, Refresh Token and the Access Token's expiration time.
+     */
+    public Future<AuthenticationResult> acquireToken(final String resource,

```

Comment:
public clients will support username password flow. you should be able to use a public client to get a token using your user account for your confidential client app (web app) as the resouce. The web app upon receiving the token can redeem is to get more tokens for another resource using on behalf of flow.